tests/pauli/test_conversion.py

```diff
@@ -28,25 +28,44 @@
     np.array([[-2, -2 + 1j, -2, -2], [-2 - 1j, 0, 0, -1], [-2, 0, -2, -1], [-2, -1, -1, 0]]),
 ]
 
+test_general_matrix = [
+    np.array([[2.5, -0.5, 1.0], [-0.5, 2.5, -1j]]),
+    np.array(np.diag([0, 0, 0, 0, 1])),
+    np.array(
+        [
+            [-2, -2 + 1j, -2, -2, -1],
+            [-2 - 1j, 0, 0, -1, -1j],
+            [-2, 0, -2, -1, 2],
+            [-2, -1, -1, 0, 2j],
+        ]
+    ),
+]
+
+test_diff_matrix = [[[-2, -2 + 1j]], [[-2, -2 + 1j], [-1, -1j]]]
+
 
 class TestDecomposition:
     """Tests the pauli_decompose function"""
 
-    @pytest.mark.parametrize("hamiltonian", [np.ones((3, 3)), np.ones((4, 2)), np.ones((2, 4))])
-    def test_wrong_shape(self, hamiltonian):
+    @pytest.mark.parametrize("hamiltonian", [np.ones((4, 2)), np.ones((2, 4))])
+    def test_wrong_shape_non_square(self, hamiltonian):
         """Tests that an exception is raised if the Hamiltonian does not have
         the correct shape"""
         with pytest.raises(
             ValueError,
-            match="The matrix should have shape",
+            match="The matrix should be square",
         ):
             qml.pauli_decompose(hamiltonian)
 
-    def test_not_hermitian(self):
-        """Tests that an exception is raised if the Hamiltonian is not Hermitian, i.e.
-        equal to its own conjugate transpose"""
-        with pytest.raises(ValueError, match="The matrix is not Hermitian"):
-            qml.pauli_decompose(np.array([[1, 2], [3, 4]]))
+    @pytest.mark.parametrize("hamiltonian", [np.ones((5, 5)), np.ones((3, 3))])
+    def test_wrong_shape_non_power_two(self, hamiltonian):
```

Comment:
We were not checking for this before?

Reply:
There was a previous function called `test_wrong_shape`, but I had to rewrite it to support including the message for `padding`. So rewrote the tests as well.